SIGSEGV in statsd monitor

Faced this error for the first time today with the quay.io/signalfx/signalfx-agent:5.3.2 docker image running on GKE:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc30 pc=0x468c45]

goroutine 242 [running]:
github.com/signalfx/signalfx-agent/pkg/monitors/statsd.(*statsDListener).Read(0xc0005018f0)
	/usr/src/signalfx-agent/pkg/monitors/statsd/statsd.go:78 +0x256
```

Not sure yet what could cause this yet as the corresponding code seems relatively trivial: https://github.com/signalfx/signalfx-agent/blob/v5.3.2/pkg/monitors/statsd/statsd.go#L78

--- a/pkg/monitors/statsd/statsd.go
+++ b/pkg/monitors/statsd/statsd.go
@@ -4,10 +4,12 @@ import (
 	"net"
 	"strconv"
 	"strings"
+	"sync"
 	"sync/atomic"
 )
 
 type statsDListener struct {
+	sync.Mutex
 	ipAddr         string
 	port           uint16
 	tcp            bool
@@ -55,10 +57,12 @@ func (sl *statsDListener) listenTCP() error {
 }
 
 func (sl *statsDListener) FetchMetrics() []*statsDMetric {
+	sl.Lock()
 	rawMetrics := make([]string, len(sl.metricBuffer))
 
 	copy(rawMetrics, sl.metricBuffer)
 	sl.metricBuffer = nil
+	sl.Unlock()
 
 	parsed := parseMetrics(rawMetrics, sl.converters, sl.prefix)
 
@@ -75,7 +79,9 @@ func (sl *statsDListener) Read() {
 	}
 
 	for data := range chData {
+		sl.Lock()
 		sl.metricBuffer = append(sl.metricBuffer, strings.Split(string(data), "\n")...)
+		sl.Unlock()
 	}
 }
 

